org.eclipse.scout.rt.ui.html/src/main/java/org/eclipse/scout/rt/ui/html/json/MultipartConfigFilter.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2010, 2024 BSI Business Systems Integration AG
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.scout.rt.ui.html.json;
+
+import java.io.IOException;
+import java.util.regex.Matcher;
+
+import org.eclipse.scout.rt.platform.Bean;
+
+import jakarta.servlet.Filter;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.ServletRequest;
+import jakarta.servlet.ServletResponse;
+import jakarta.servlet.http.HttpServletRequest;
+
+/**
+ * <p>
+ * Filter class to remove a possibly set multipart config from {@link HttpServletRequest} if this filter determines it
+ * is not necessary.
+ * </p>
+ * <p>
+ * This is necessary as otherwise reading parameters may consume the request payload (to determine parameters) which
+ * makes other body reading methods e.g. {@link HttpServletRequest#getInputStream()} unusable.
+ * </p>
+ *
+ * @see <a href="https://github.com/jetty/jetty.project/issues/1175">jetty/jetty.project issue #1175</a>
+ * @see HttpServletRequest#getParameter(String)
+ */
+@Bean
+public class MultipartConfigFilter implements Filter {
+
+  @Override
+  public void doFilter(ServletRequest req0, ServletResponse resp, FilterChain chain) throws IOException, ServletException {
+    removeConfigIfNotNeeded(req0);
+    chain.doFilter(req0, resp);
+  }
+
+  protected void removeConfigIfNotNeeded(ServletRequest req0) {
+    if (!(req0 instanceof HttpServletRequest)) {
+      return;
+    }
+
+    HttpServletRequest req = (HttpServletRequest) req0;
+
+    if (!needsConfig(req)) {
+      // see org.eclipse.jetty.server.Request.__MULTIPART_CONFIG_ELEMENT (literal value)
+      req.removeAttribute("org.eclipse.jetty.multipartConfig");
```

Comment:
Remove config here or add it explicitly only for /upload Path here and remove in default UiServlet registration?
I personally would prefer to remove the global multipart config and just add it here for file upload paths.

Reply:
Changed behavior from remove to add.